Created

Uuqlele opened the pull request.

Commit:
Created

DaemOn66613 commented on CSharpCourse/Game/Character.cs:
```diff
@@ -0,0 +1,104 @@
+        public int Damage { get; set; }
+        public int Exp = 0;
+        public int Level = 1;
+        public int Role = 0;
```
Role должна быть не int, а Roles.

DaemOn66613 commented on CSharpCourse/Game/Character.cs:
```diff
@@ -0,0 +1,104 @@
+        {
+            Console.Clear();
+            int role = Menu.CreateMenu("Выберете персонажа (хп/урон):",
+                new List<string>() { "1. Воин (25/4)",
```
Я бы сделал базовый класс персонажа, а настройки хп и проч. вынес бы в класс, который наследуется от базового, т.е. есть класс Character, который содержит характеристики и есть 3 класса персонажей, которые наследуют класс Character и переопределяют кол-во хп, урон, способность и т.д.

DaemOn66613 replied:
Или пока не заморачиваться с разными моделями и классами просто сделать 3 конструктора, в который передаешь выбранный класс, а не делать выбор в самом конструкторе, для этого есть меню

Uuqlele replied:
> Или пока не заморачиваться с разными моделями и классами просто сделать 3 конструктора, в который передаешь выбранный класс, а не делать выбор в самом конструкторе, для этого есть меню

Не совсем понимаю, как мне это сделать в данной ситуации. Тут все сделано так, что персонаж начинает создаваться когда пользователь выбирает пункт меню, индекс пункта летит в свитч, откуда уже и задаются конкретные параметры персонажу. Как мне с помощью индекса меню от пользователя выбрать конкретный конструктор? Нужно конструкторы в кейсах свитча определить?

DaemOn66613 commented on CSharpCourse/Game/Character.cs:
```diff
@@ -0,0 +1,104 @@
+            return monster.HitPoints;
+        }
+
+        public void UseAbility(ref Monster monster, ref Character theHero)
```
в метод использования атаки и способности не надо передавать данные персонажа, ты сможешь обратиться к ним через this, т.к. работать будешь с объектом класса твоего персонажа. Аналогично, в расчете атаки монстра нужен только объект персонажа.

Uuqlele replied:
> в метод использования атаки и способности не надо передавать данные персонажа, ты сможешь обратиться к ним через this, т.к. работать будешь с объектом класса твоего персонажа. Аналогично, в расчете атаки монстра нужен только объект персонажа.

Я почему-то сразу же не додумался, поправил, но возник ещё вопрос: а нужны ли вообще this в данном случае? Классы не статические, поля имеются только у объектов, получается даже this нет необходимости использовать.

Commit:
Роль персонажа теперь Role, а не int;
Убрана лишняя передача данных в методы использования способности и атаки героев и монстров.
